document-generation/igce-document.ts

```diff
@@ -0,0 +1,270 @@
+  updateSummaryFormulas(summarySheetAdditionalRows, summarySheet);
+
+  // exceljs does not support text boxs or shapes and the signature text box
+  // is removed during generation of the IGCE document
+  // see https://github.com/exceljs/exceljs/issues/1147
+  // Add in signature rows to replace the removed text box
+  const summarySheetSignatureRow = 31 + summarySheetAdditionalRows;
+  summarySheet.getCell(`A${summarySheetSignatureRow}`).value = "Preparer Name: _____________________";
+  summarySheet.getCell(`A${summarySheetSignatureRow + 1}`).value = "Title: _____________________________";
+  summarySheet.getCell(`A${summarySheetSignatureRow + 2}`).value = "Email: _____________________________";
+  summarySheet.getCell(`A${summarySheetSignatureRow + 4}`).value = "Signature: _________________________";
+
+  const headers = {
+    "Content-Type": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+    "Content-Disposition": `attachment; filename=IndependentGovernmentCostEstimate.xlsx`,
+  };
+
+  const buffer = (await workbook.xlsx.writeBuffer()) as Buffer;
+  logger.info("IGCE document generated");
+
+  return new ApiBase64SuccessResponse(buffer.toString("base64"), SuccessStatusCode.OK, headers);
+}
+
+function updatePeriodFormulas(row: Row, rowNumber: number, sheet: Worksheet): void {
+  const rowValues: any = row.values;
+  const cellId = `I${rowNumber}`;
+  if (rowValues[9] && String(rowValues[9].formula).includes("&C")) {
+    const formula: string = rowValues[9].formula;
+    const splitFormula = formula.split("&C");
+    const newFormula = splitFormula[0] + `&C${rowNumber}` + splitFormula[1].slice(2);
+    const cell = sheet.getCell(cellId);
+    sheet.getCell(cellId).value = { ...cell, formula: newFormula };
+  }
+
+  if (String(rowValues[7]).includes("Total Price")) {
+    const totalCell = sheet.getCell(cellId);
+    sheet.getCell(cellId).value = { ...totalCell, formula: `=SUM(I28:I${rowNumber - 1})` };
+  }
+}
+
+function updateSummaryFormulas(numberOfAddedRows: number, sheet: Worksheet): void {
+  const initialSumRow = 22;
+  const summaryStartRow = initialSumRow + numberOfAddedRows;
+  const columns = ["C", "D", "E", "F", "G", "H", "I", "J"];
+
+  // removed original calculations due to an error with master and shared formula types
+  // added back in the original formulas to calculate totals
+  const summaryCalculations = sheet.getRows(summaryStartRow, 3);
+  summaryCalculations?.forEach((row, index) => {
+    columns.forEach((col) => {
+      const cellId = `${col}${row.number}`;
+      const cell = sheet.getCell(cellId);
+      let updatedFormula;
+      if (index === 0) {
+        updatedFormula = `=SUM(${col}6:${col}${row.number - 3})`;
+      } else if (index === 1) {
+        updatedFormula = `=${col}${row.number - 1}*.0225`;
+      } else if (index === 2) {
+        updatedFormula = `=SUM(${col}${row.number - 2}:${col}${row.number - 1})`;
+      } else {
+        updatedFormula = "";
+      }
```

Comment:
This would also make a good `switch`

Reply:
If you're doing `===` comparisons against literals or an `enum`, then you probably want a `switch`. If you're doing complex logic based on function calls or whatever, `if` is probably better. And I think in our cases here, we've mostly got literals and enums